### ISSUE TITLE
How to add a job with multiple stops before the final drop-off.

### ISSUE THREAD

**Issue by albertsallesmosaic** [VROOM-Project/vroom#924]
Hello! First of all, thanks for this amazing project. I've been looking at the docs and different issues but I can't find an answer for my problem, maybe I am not understanding something correctly.

My problem consists on different tasks, each task can have one of this two forms:
- A vehicle must pick up a package at a certain place, deliver it to several locations and finally drop it off empty in a depot.
` PickUp -> Location1 -> ... -> LocationN -> Depot`
- Or the other way around: pick up the package empty, go to several locations and leave the package full at the final destination.
`Depot -> Location1 -> ... -> LocationN -> Destination`

Each task can only be performed by one vehicle, because the package is treated as a whole. The order of the LocationX does not matter, only the pickup and final destination are required to be in that order depending on the task.

I've been trying to define each task as different jobs but the problem is that the package must be picked up before the deliveries and the final destination must be at the end. And looking at the docs it seems that a shipment has only one pickup and one delivery location. Is it possible to solve this problem with VROOM? If not, is there any workaround to solve it with less constraints?

The goal in the end will be to determine how many vehicles are needed to accomplish all the tasks, the fewer the better.

I appreciate your time and effort!

**Comment by jcoupey** [comment]
> a shipment has only one pickup and one delivery location

Yes, so there is no way to "link" more than two tasks together to make sure they appear in the same route and in the right order.

> is there any workaround to solve it with less constraints?

I don't think so if you expect to group and interleave several such chain of tasks in a single vehicle route.

**Comment by albertsallesmosaic** [comment]
I can imagine that it would be possible if I defined each task as a simple pickup-and-delivery shipment that starts at `PickUp` location and ends at `Depot` location but with custom stops in between.

In this case the routing machine would have to decide the best path to do this task: 
`PickUp -> LocationX -> (N times) -> LocationY -> Depot`
And they would be treated as normal shipments from one location to another.

So is it possible to modify the code to implement such scenario? I'd be down to implement it if it's not very hard.

Thanks!

**Comment by albertsallesmosaic** [comment]
Appreciate your responses!

Certainly for an arbitrary number of intermediate stops for one shipment would increase complexity considerably. In the case of my problem, the max number of intermediate stops would be 5. So `N = 5` in my previous examples. I was asking for an arbitrary number `N` in case it was fairly simple.

In summary, one task = one shipment, and each shipment has maximum 5 intermediate stops. Do you still think it is too complex to implement and run?

Thank you!!

**Comment by jcoupey** [comment]
>Do you still think it is too complex to implement and run?

It depends  if you're prepared to deep into the internal logic and change a lot of it. Like I said, it would touch every part of the solving, from the heuristic that define initial solutions to the local search process where you'd have to adjust all relevant operators.

In term of benefit/effort ratio, you may be better off using another solving engine that is flexible enough to allow adding such a constraint in an easy way.

**Comment by albertsallesmosaic** [comment]
Understood! Thanks a lot for your time. If you could recommend me another solving engine that'd be great, otherwise have a great day.

**Comment by jcoupey** [comment]
Closing as answered.
